src/hooks.ts

```diff
@@ -35,17 +36,20 @@ export const useExternalPlaybackAvailability = (
 };
 
 export const useAirplayConnectivity = () => {
-  const [isAirplayConnected, setIsAirplayConnected] = useState(false);
+  const routes = useAirplayRoutes();
+  return routes.filter(route => route.portType === 'AirPlay').length > 0;
+};
+
+export const useAirplayRoutes = () => {
+  const [airplayRoutes, setAirplayRoutes] = useState<AVSessionRoute[]>([]);
 
   useEffect(() => {
-    const subscription = onAirplayConnectivityChanged(setIsAirplayConnected);
+    const subscription = onAirplayConnectivityChanged(setAirplayRoutes);
 
-    AirplayConnectivityContext?.fetchAirplayConnectivity().then(
-      setIsAirplayConnected,
-    );
+    AirplayConnectivityContext?.fetchAirplayRoutes().then(setAirplayRoutes);
 
     return subscription.remove.bind(subscription);
   }, []);
 
-  return isAirplayConnected;
+  return airplayRoutes;
```

Comment:
Would be better if new hooks are created name `useAirplayRoutes` & `useAVAudioSessionRoutes`. So that there's no breaking change, the existing `useAirplayConnectivity` can just check length returned by the new hooks

Reply:
I've renamed the new hook to `useAVAudioSessionRoutes` and created one based on that for `useAirplayRoutes`.